android/ui/src/main/java/com/trafi/ui/Badge.kt

```diff
@@ -0,0 +1,258 @@
+                    }
+            )
+        }
+    }
+}
+
+
+@OptIn(ExperimentalLayout::class)
+@Composable
+private fun StackedBadge(
+    modifier: Modifier = Modifier,
+    badge: BadgeInfo?,
+    alternativeBadges: List<BadgeInfo>,
+    vector: ImageVector?,
+    subbadge: ImageVector? = null,
+    badgeType: BadgeType
+) {
+    ConstraintLayout {
+        val altBadgesSublist =
+            if (alternativeBadges.size > 2) alternativeBadges.subList(0, 2) else alternativeBadges
+        altBadgesSublist.forEachIndexed { index, smallScheduleBadge ->
+            Surface(
+                color = smallScheduleBadge.color,
+                contentColor = smallScheduleBadge.textColor ?: constants.defaultContentColor,
+                border = BorderStroke(constants.borderWidth, color = constants.defaultContentColor),
+                shape = RoundedCornerShape(getBadgeRounding(badgeType)),
+                modifier = modifier.height(getBadgeHeight(badgeType) + (index * 4).dp)
+                    .padding(top = (index * 4).dp)
+            ) {
+                BadgeFiller(
+                    badgeType = badgeType,
+                    vector = vector,
+                    // This is not a mistype, it is a lifehack to make badges same size
+                    text = badge?.text,
+                    isHiddenLayoutFiller = true,
+                    isStackedBadge = true
+                )
+            }
+        }
+        Surface(
+            color = badge?.color ?: Color.Black,
+            contentColor = badge?.textColor ?: constants.defaultContentColor,
+            border = BorderStroke(constants.borderWidth, color = constants.defaultContentColor),
+            shape = RoundedCornerShape(getBadgeRounding(badgeType)),
+            modifier = modifier
+                .height(getBadgeHeight(badgeType) + (altBadgesSublist.size * 4).dp + (constants.borderWidth * 2))
+                .padding(top = (altBadgesSublist.size * 4).dp)
+        ) {
+            BadgeFiller(
+                badgeType = badgeType,
+                vector = vector,
+                text = badge?.text,
+                isHiddenLayoutFiller = false,
+                isStackedBadge = true
+            )
+        }
+        subbadge?.let {
+            Image(
+                subbadge,
+                modifier = Modifier
+                    .width(constants.subBadgeIconWidth)
+                    .height(constants.subBadgeIconHeight)
+                    .padding(end = 3.dp, bottom = 3.dp)
+                    .constrainAs(createRef()) {
+                        bottom.linkTo(parent.bottom)
+                        top.linkTo(parent.bottom)
+                        start.linkTo(parent.end)
+                        end.linkTo(parent.end)
+                    }
+            )
+        }
+    }
+}
+
+@Composable
+private fun BadgeFiller(
+    badgeType: BadgeType,
+    vector: ImageVector?,
+    text: String?,
+    isHiddenLayoutFiller: Boolean = false,
```

Comment:
Optional parameters with default values should be declared after mandatory parameters.